src/components/report-viewer/report-viewer.tsx

```diff
@@ -96,6 +99,30 @@ export default function ReportViewer({ report, reportType }: ReportViewerProps)
         setHighlightedReportId(data.parentId);
     };
 
+    const handleSelectedItem = useCallback(
+        (report: ReportItem) => {
+            if (selectedReportId !== report.id) {
+                setSelectedReportId(report.id);
+                setSelectedReportType(reportTreeData.current[report.id].type);
+            }
+        },
+        [selectedReportId]
+    );
+
+    const handleExpandItem = useCallback(
+        (node: ReportItem) => {
+            if (node.collapsed) {
```

Comment:
collapsed handling code should be moved to the virtaulized-treeview component if it's not too complicated